src/dmd/mars.d

```diff
@@ -360,8 +360,33 @@ private int tryMain(size_t argc, const(char)** argv)
     }
     if (files.dim == 0)
     {
-        usage();
-        return EXIT_FAILURE;
+        // -X without arguments is the JSON analog of --version
+        if (global.params.doJsonGeneration)
+        {
+            OutBuffer buf;
+            json_info(&buf);
+            buf.writestring("\n");
+            // write to stdout for -X and -Xf=-
+            if (!global.params.jsonfilename || strcmp(global.params.jsonfilename, "-") == 0)
```

Comment:
If wanted I can split this if statement and print the manual for `-X` (previous behavior).